### DISCUSSION TITLE
How can I use Close price in MESA Adaptive Moving Average?

### DISCUSSION THREAD

**Discussion by allcasper99** [DaveSkender/Stock.Indicators#500] [Q&A] [answered]
Hello there,
I checked the MESA Adaptive Moving Average indicator. It works very well thanks.

I have a question.
I want to use the closing price instead of hl2 as the source, but I couldn't see such an option. Could you help?

**Comment by DaveSkender** [comment, marked as answer]
As you've mentioned, the [formula for MAMA](http://mesasoftware.com/papers/MAMA.pdf) calls for the use of `Price=(H+L)/2`.  If you want to use `Close` price instead, the workaround here would be to convert your quote history to accommodate so `(H+L)/2 = Close`.

```csharp
var modifiedQuotes = quotes
   .Select(x => new Quote
   {
       Date = x.Date,
       High = x.Close,
       Low = x.Close,
       Close = x.Close
   });

var results = modifiedQuotes.GetMama(..);
